content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -102,11 +102,103 @@ export const problems = {
 
 <FocusProblem problem={problems.maxsum} />
 
+### Solution - Max Subarray Sum
+
+<Spoiler title="Solution">
+
+Consider the prefix sum array $p$. The subarray sum $a_i \dots a_{j-1}$ where $i\leq j$ is $p[j]-p[i]$, so we are looking for the maximum possible value of $p[j]-p[i]$ over $0 \leq i \leq j \leq N$. 
+
+For a fixed right bound $j$, the maximum subarray sum is
+
+$$
+p[j]-\min_{i\leq j}{p[i]}
+$$
+
+Thus, we can keep a running minimum to store $\min\limits_{i\leq j}{p[i]}$ as we iterate through $i$. This yields the maximum subarray sum for each possible right bound, and the maximum over all these values is our answer.
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl pfx(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		pfx[i] = pfx[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = pfx[0], min_prefix_sum = pfx[0];
+	for(int i=1; i<=n; i++){
+		// Max subarray sum is the maximum difference between two prefix sums
+		max_subarray_sum = max(max_subarray_sum, pfx[i] - min_prefix_sum);
+		min_prefix_sum = min(min_prefix_sum, pfx[i]);
+	}
+	cout << max_subarray_sum << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
+
 This problem has a solution known as [Kadane's Algorithm](https://en.wikipedia.org/wiki/Maximum_subarray_problem#Kadane's_algorithm). Please don't use that solution; try to solve it with prefix sums.
 
-<Spoiler title="Why are the two methods equivalent?">
+</Warning>
+
+<Spoiler title = "Alternative Solution: Kadane's Algorithm">
+
+Kadane's Algorithm finds the largest subarray sum using a greedy method. Check this
+[video tutorial](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s) for more details.
```

Comment:
a link to wiki is alr linked so just remove the vid.

Reply:
Alright... Also, the test worked earlier, but after changes, it failed :(...